extensions/parquet/table/src/test/java/io/deephaven/parquet/table/TestParquetTools.java

```diff
@@ -580,4 +611,574 @@ public void testNoDictionaryOffset() {
                 DateTimeUtils.epochNanos((Instant) withNullsAndMissingOffsets.getColumnSource("CREATE_DATE").get(0)));
         assertTableEquals(withNullsAndMissingOffsets, clean);
     }
+
+    @Test
+    public void testWriteParquetFieldIds() throws NoSuchAlgorithmException, IOException {
+        final int BAZ_ID = 111;
+        final int ZAP_ID = 112;
+        final String BAZ = "Baz";
+        final String ZAP = "Zap";
+        final ColumnDefinition<Long> bazCol = ColumnDefinition.ofLong(BAZ);
+        final ColumnDefinition<?> zapCol = ColumnDefinition.of(ZAP, Type.stringType().arrayType());
+        final TableDefinition td = TableDefinition.of(bazCol, zapCol);
+        final Table expected = newTable(td,
+                longCol(BAZ, 99, 101),
+                new ColumnHolder<>(ZAP, String[].class, String.class, false, new String[] {"Foo", "Bar"},
+                        new String[] {"Hello"}));
+        final File file = new File(testRoot, "testWriteParquetFieldIds.parquet");
+        {
+            // Writing down random parquet column names that we _don't_ keep a reference to. This way, the only way we
+            // can successfully resolve them is by field id.
+            final ParquetInstructions writeInstructions = ParquetInstructions.builder()
+                    .setFieldId(BAZ, BAZ_ID)
+                    .setFieldId(ZAP, ZAP_ID)
+                    .addColumnNameMapping("Some Random Parquet Column Name", BAZ)
+                    .addColumnNameMapping("ABCDEFG", ZAP)
+                    .build();
+            ParquetTools.writeTable(expected, file.getPath(), writeInstructions);
+        }
+
+        // This is somewhat fragile, but has been manually verified to contain the field_ids that we expect.
+        // We may want to consider more explicit tests that verify our writing logic is consistent, as it would be good
+        // to know whenever our serialization changes in any way.
+        assertEquals("c21f162b2c186d0a95a8d2302c9ed3fab172747b45501b2ee5e5bb04b98e92e0", sha256sum(file.toPath()));
+
```

Comment:
I think it would be nice to read the file, and test that the metadata has the field IDs we expect.